docs/topics/proc_exposing-application-metrics-using-prometheus-on-node-js.adoc

```diff
@@ -0,0 +1,129 @@
+
+. Append the following snippet to the `prometheus.yml` file to make Prometheus automatically collect metrics from your application:
++
+[source,yaml]
+----
+  - job_name: nodejs
+     static_configs:
+     - targets: ['localhost:8888']
+----
++
+The default behavior of {Node}-based applications is to expose metrics at the `/metrics` endpoint.
+This is also what Prometheus expects.
+--
+
+* The Prometheus server running on `localhost` using Docker:
++
+--
+[source,bash,opts="nowrap"]
+----
+# docker run -it --rm --network host -p 9090:9090 -v $PWD/prometheus.yml:/etc/prometheus/prometheus.yml prom/prometheus
+----
+
+Wait until the `Server is ready to receive web requests` message is displayed in the console.
+
+Include the the `--network host` option to allow Prometheus to connect to your application.
+--
+
+.Procedure
+
+. Install the `prom_client` npm package.
++
+.pom.xml
+[source,bash]
+----
+$ npm install prom-client express body-parser --save
+----
+
+. Create an example {NodeJS} application and save it in a file with the file name `app.js` .
++
+--
+[source,javascript]
+----
+'use strict';
+const path = require('path');
+const express = require('express');
+const bodyParser = require('body-parser');
+const client = require('prom-client');
+const histogram = new client.Histogram({
+  name: 'hello',
+  help: 'Duration of HTTP requests in ms',
+  labelNames: ['status_code'],
+  buckets: [0.1, 5, 15, 50, 100, 500]
+});
+const app = express();
+const port = 8888;
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+app.use('/api/greeting', (request, response) => {
+  const end = histogram.startTimer();
+  const name = request.query ? request.query.name : undefined;
+  response.send({content: `Hello, ${name || 'World!'}`});
+  end();
+});
+
+app.get('/metrics', (req, res) => {
+  res.set('Content-Type', client.register.contentType);
+  res.end(client.register.metrics());
+})
+
+app.listen(port, () => console.log(`Example app listening on port ${port}!`))
+
+----
+--
+
+. Launch your application:
++
+[source,bash,opts="nowrap"]
+----
+$ node app.js
+----
+
+. Invoke the traced endpoint several times:
++
+[source,bash,opts="nowrap"]
+----
+$ curl http://localhost:8888/api/greeting
+{"content":"Hello, World!"}
+----
+
+. Wait at least 15 seconds for the collection to happen, and view the metrics in the Prometheus web UI:
++
+--
+. Open the Prometheus UI at link:http://localhost:9090/[http://localhost:9090/^] and type `hello` into the _Expression_ box.
+. From the suggestions, select for example `application:hello_count` and click the _Execute_ button.
+. In the table below, you can see the number of times you invoked the resource method.
+// this metric seems to be unavailable for the node app.
+//. Alternatively, select `application:hello_time_mean_seconds` to see the mean time of all the invocations.
+
+//all user-defined metrics should have this
+Note that all metrics you created are prefixed with `application:`.
```

Comment:
So the Node.js Prometheus library does that on its own? Cool.

Reply:
@Ladicek Not sure it does, gonna have to check with Daniel to be certain.

Reply:
@inoxx03 @Ladicek see line 77 of the node application. That line sets `application:` as the prefix.